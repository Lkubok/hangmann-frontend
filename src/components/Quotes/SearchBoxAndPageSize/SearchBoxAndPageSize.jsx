import React, { PureComponent } from "react";
import { connect } from "react-redux";
import {
  changeFilter,
  changeSortedQuotes,
  changePageSize
} from "../../../actions/quoteActions";
import * as selectors from "../../../reducers/selectors";
import "./SearchBoxAndPageSize.scss";

export class SearchBoxAndPageSize extends PureComponent {
  handleChange = e => {
    const { value, name } = e.target;
    const { changePageSize, changeFilter, changeSortedQuotes } = this.props;
    if (name === "pageSize") changePageSize(parseInt(value));
    else {
      changeFilter(value);
      changeSortedQuotes(this.props.quotes);
    }
  };
  render() {
    return (
      <>
        <input
          type="text"
          onChange={this.handleChange}
          value={this.props.filterQuery}
          className="quote-query-input"
          placeholder="Type search query.."
        />
        <select
          name="pageSize"
          value={this.props.pageSize}
          onChange={this.handleChange}
          className="quote-page-size"
        >
          <option value="10">--- chose limit per page ---</option>
          <option value="5">5</option>
          <option value="10">10</option>
          <option value="15">15</option>
          <option value="20">20</option>
          <option value="30">30</option>
          <option value="50">50</option>
          <option value="100">100</option>
        </select>
      </>
    );
  }
}

const mapStateToProps = state => ({
  filterQuery: state.quotesReducer.filterQuery,
  pageSize: state.quotesReducer.pageSize,
  quotes: selectors.getSortedQuotes(state)
});

const mapDispatchToProps = {
  changeFilter,
  changeSortedQuotes,
  changePageSize
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBoxAndPageSize);
