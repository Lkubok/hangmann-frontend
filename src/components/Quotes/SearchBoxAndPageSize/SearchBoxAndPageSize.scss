@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.quote-query-input {
  @include input-quote-header;
  flex-basis: 25%;
}

.quote-page-size {
  @include input-quote-header;

  option {
    text-align: center;
  }
}
