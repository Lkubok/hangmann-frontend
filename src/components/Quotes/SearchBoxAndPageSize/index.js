import SearchBoxAndPageSize from "./SearchBoxAndPageSize";
export default SearchBoxAndPageSize;
