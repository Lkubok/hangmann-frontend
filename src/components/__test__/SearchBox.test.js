import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { SearchBoxAndPageSize } from "../Quotes/SearchBoxAndPageSize/SearchBoxAndPageSize";
Enzyme.configure({ adapter: new Adapter() });

describe("Search Box Component", () => {
  it("Should match snapshot after mount", () => {
    const wrapper = shallow(<SearchBoxAndPageSize />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should trigger change filter query on input change", () => {
    const wrapper = shallow(<SearchBoxAndPageSize />);
    const spyFunction = jest.fn();
    const event = {
      target: {
        value: "value in input",
        name: null
      }
    };
    wrapper.instance().handleChange = spyFunction;
    wrapper.instance().changeFilter = spyFunction;
    wrapper.update();
    wrapper.find("#filterQuery").simulate("change", event);
    expect(spyFunction).toHaveBeenCalled();
  });
});
