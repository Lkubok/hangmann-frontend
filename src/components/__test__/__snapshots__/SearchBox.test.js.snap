// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box Component Should match snapshot after mount 1`] = `
<Fragment>
  <input
    className="quote-query-input"
    id="filterQuery"
    onChange={[Function]}
    placeholder="Type search query.."
    type="text"
  />
  <select
    className="quote-page-size"
    name="pageSize"
    onChange={[Function]}
  >
    <option
      value="10"
    >
      --- chose limit per page ---
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="10"
    >
      10
    </option>
    <option
      value="15"
    >
      15
    </option>
    <option
      value="20"
    >
      20
    </option>
    <option
      value="30"
    >
      30
    </option>
    <option
      value="50"
    >
      50
    </option>
    <option
      value="100"
    >
      100
    </option>
  </select>
</Fragment>
`;
